File: tripplanning-ms/tripplanning-lib/src/main/java/com/flashex/tripplanningmicroservice/lib/model/TripItinerary.java
package com.flashex.tripplanningmicroservice.lib.model;

import lombok.*;

import java.util.Date;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class TripItinerary {

    private String tripItineraryId;
    private List<Packet> packets;
    private Date plannedStartTime;
    private Date plannedEndTime;
    private float plannedTotalDistance;
    private Vehicle vehicle;
    private float tripExpense;
    private float occupiedVolume;

}

File: common-ms/flashex-webapp/angular-boiler-plate/src/app/trip-management/interfaces/trip-itinerary.ts
import { IPacket } from '../interfaces/Packet';
import { IVehicle } from '../interfaces/vehicle';

export interface IItinerary {
  tripItineraryId: number;
  packets: IPacket[];
  plannedStartTime: Date;
  plannedEndTime: Date;
  plannedTotalDistance: number;
  vehicle: IVehicle;
  tripExpense: number;
  occupiedVolume: number;
  originAddress: string;
  algorithm: string;
}
